### PR TITLE
fix for getSchema when using "-" in name

### PR DESCRIPTION
**Pull request by rene-ye**
Fix for issue #715 by @fabiocorneti

### CODE REVIEW

**Comment by fabiocorneti** [comment]
Apologies for not opening the PR myself, was busy at work :)

File: src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java [hunk 1]
```diff
@@ -175,6 +179,106 @@ public void testDBSchema() throws SQLException {
...
+                hasResults = true;
+                final String schemaName = rs.getString(1);
+                final String catalogName = rs.getString(2);
+                assertTrue(!StringUtils.isEmpty(schemaName), schemaEmptyFormat.format(schemaMsgArgs));
+                assertTrue(!StringUtils.isEmpty(catalogName), schemaEmptyFormat.format(catalogMsgArgs));
+                assertEquals(schemaName, schemaMsgArgs[0]);
+                assertEquals(catalogName, catalogMsgArgs[0]);
+            }
+
+            final MessageFormat atLeastOneFoundFormat = new MessageFormat(TestResource.getResource("R_atLeastOneFound"));
```
Comment (cheenamalhotra) [inline src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java]: Everything else looks okay, there's just too many `final` objects here in both new methods, which don't really have a purpose. Let's keep it consistent with other tests.

File: src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java [hunk 2]
```diff
@@ -175,6 +179,106 @@ public void testDBSchema() throws SQLException {
...
+            assertTrue(hasResults, atLeastOneFoundFormat.format(schemaMsgArgs));
+        } finally {
+            if (dropDatabase) {
+                connection.createStatement().execute(String.format("DROP DATABASE [%s]", testCatalog));
```
Comment (cheenamalhotra) [inline src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java]: You're using the generic connection and creating statement, which is not closed (same for other method), you could use _Utils_ functions to drop database for consistency and that clear up objects as well.

File: src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java [hunk 3]
```diff
@@ -175,6 +179,106 @@ public void testDBSchema() throws SQLException {
...
+    public void testDBSchemasForDashedCatalogNameWithPattern() throws SQLException {
+        final UUID id = UUID.randomUUID();
+        final String testCatalog = "dash-catalog"+id;
+        final String testSchema = "some-schema"+id;
+        boolean dropDatabase = false;
+
+        try (final Connection dashConn = DriverManager.getConnection(connectionString);
+             final Statement dashStatement = dashConn.createStatement()) {
+
+            connection.createStatement().execute(String.format("CREATE DATABASE [%s]", testCatalog));
```
Comment (cheenamalhotra) [inline src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java]: Unclosed Statement, either close this Statement, or use `dashStatement` instead.

File: src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java [hunk 4]
```diff
@@ -175,6 +179,106 @@ public void testDBSchema() throws SQLException {
...
+     * Tests that the catalog parameter containing - is escaped by
+     * {@link SQLServerDatabaseMetaData#getSchemas(String catalog, String schemaPattern)}.
+     * @throws SQLException
+     */
+    @Test
+    public void testDBSchemasForDashedCatalogNameWithPattern() throws SQLException {
+        final UUID id = UUID.randomUUID();
+        final String testCatalog = "dash-catalog"+id;
+        final String testSchema = "some-schema"+id;
+        boolean dropDatabase = false;
```
Comment (cheenamalhotra) [inline src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java]: This varible can be avoided if you'd use _Utils_ drop functions, that check for existence of objects in sql.

File: src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java (hunk 3 above)
Comment (rene-ye) [reply src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java]: Created with try-with-resources, auto closed at the end of the blocks.

File: src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java (hunk 4 above)
Comment (rene-ye) [reply src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java]: Utils is able to drop (if exists) stored procedures and tables. There doesn't seem to be one to handle databases.

File: src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java (hunk 1 above)
Comment (rene-ye) [reply src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java]: Removed all instances of `final`.

File: src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java [hunk 5]
```diff
@@ -175,6 +179,100 @@ public void testDBSchema() throws SQLException {
...
+    @Test
+    public void testDBSchemasForDashedCatalogName() throws SQLException {
+        UUID id = UUID.randomUUID();
+        String testCatalog = "dash-catalog"+id;
+        String testSchema = "some-schema"+id;
+
+        try (Connection dashConn = DriverManager.getConnection(connectionString);
+             Statement dashStatement = dashConn.createStatement()) {
+
+            connection.createStatement().execute(String.format("CREATE DATABASE [%s]", testCatalog));
```
[the remark this answers is not in the document]
Comment (rene-ye) [reply src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java]: The unique id makes this unlikely, and Utils does not have a dropDatabaseIfExists functionality (only stored procedure and table). However, I will adjust the code to check for this as it's a good habit.

[the remark this answers is not in the document]
Comment (ulvii) [reply src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java]: `dropDatabaseIfExists ` is already on dev branch.

File: src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java [hunk 6]
```diff
@@ -175,6 +179,100 @@ public void testDBSchema() throws SQLException {
...
+                }
+            }
+
+            MessageFormat atLeastOneFoundFormat = new MessageFormat(TestResource.getResource("R_atLeastOneFound"));
+            assertTrue(hasResults, atLeastOneFoundFormat.format(schemaMsgArgs));
+
+            MessageFormat dashCatalogFormat = new MessageFormat(TestResource.getResource("R_atLeastOneFound"));
+            assertTrue(hasDashCatalogSchema, dashCatalogFormat.format(new Object[] {testSchema}));
+        } finally {
+            connection.createStatement().execute("IF EXISTS (SELECT name FROM sys.databases WHERE name = N'" + testCatalog + "') " +
```
Comment (ulvii) [inline src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java]: Replace this with the method from Utils class too.

File: src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java [hunk 7]
```diff
@@ -7,8 +7,10 @@
  */
```
Comment (ulvii) [inline src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java]: Please add license header.

File: src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java [hunk 8]
```diff
@@ -175,6 +179,107 @@ public void testDBSchema() throws SQLException {
...
+     * Tests that the catalog parameter containing - is escaped by {@link SQLServerDatabaseMetaData#getSchemas(String catalog, String schemaPattern)}.
+     * 
+     * @throws SQLException
+     */
+    @Test
+    public void testDBSchemasForDashedCatalogName() throws SQLException {
+        UUID id = UUID.randomUUID();
+        String testCatalog = "dash-catalog" + id;
+        String testSchema = "some-schema" + id;
+        Statement stmt = connection.createStatement();
```
Comment (ulvii) [inline src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java]: You can move this into try with resources.

Comment (rene-ye) [reply src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java]: Cannot resolve variables declared in try-with-resources in a finally block, and a finally block is used to drop the created db.

**Pull request merged**